cmd/loadbalancer/loadbalancer_create.go

```diff
@@ -0,0 +1,414 @@
+	var network *civogo.Network
+	if lbNetwork != "" {
+		if lbNetwork == "default" {
+			_, err := client.GetDefaultNetwork()
+			if err != nil {
+				utility.Error("Error fetching default network: %s", err)
+				os.Exit(1)
+			}
+		} else {
+			_, err := client.FindNetwork(lbNetwork)
+			if err != nil {
+				utility.Error("Error finding network: %s", err)
+				os.Exit(1)
+			}
+		}
+		configLoadBalancer.NetworkID = lbNetwork
+	}
+	if lbCreateFirewall == "" {
+		configLoadBalancer.FirewallRules = "80;443"
+	} else {
+		configLoadBalancer.FirewallRules = lbCreateFirewall
+	}
+
+	if lbExistingFirewall != "" {
+		if lbCreateFirewall != "" {
+			utility.Error("You can't use --create-firewall together with --existing-firewall flag")
+			os.Exit(1)
+		}
+
+		ef, err := client.FindFirewall(lbExistingFirewall)
+		if err != nil {
+			utility.Error("Unable to find existing firewall %q: %s", lbExistingFirewall, err)
+			os.Exit(1)
+		}
+
+		if ef.NetworkID != network.ID {
+			utility.Error("Unable to find firewall %q in network %q", ef.ID, network.Label)
+			os.Exit(1)
+		}
+
+		configLoadBalancer.FirewallID = ef.ID
+		configLoadBalancer.FirewallRules = ""
+	}
+}
+
+func setLoadBalancerOptions(configLoadBalancer *civogo.LoadBalancerConfig) {
+	if lbAlgorithm != "" {
+		configLoadBalancer.Algorithm = lbAlgorithm
+	}
+
+	if lbExternalTrafficPolicy != "" {
+		configLoadBalancer.ExternalTrafficPolicy = lbExternalTrafficPolicy
+	}
+
+	if lbSessionAffinity != "" {
+		configLoadBalancer.SessionAffinity = lbSessionAffinity
+	}
+
+	if lbSessionAffinityConfigTimeout != 0 {
+		configLoadBalancer.SessionAffinityConfigTimeout = int32(lbSessionAffinityConfigTimeout)
+	}
+
+	if common.RegionSet != "" {
+		configLoadBalancer.Region = common.RegionSet
+	}
+}
+
+func setLoadBalancerBackends(configLoadBalancer *civogo.LoadBalancerConfig) error {
+	var configLoadBalancerBackend []civogo.LoadBalancerBackendConfig
+
+	for _, backend := range lbBackends {
+		// Replace pipes with commas to match expected format
+		backend = strings.ReplaceAll(backend, "|", ",")
+
+		// Parse backend string into a key-value map
+		data, err := parseBackendString(backend)
+		if err != nil {
+			return fmt.Errorf("invalid backend entry: %s", err)
+		}
+
+		// Validate required fields and convert them
+		ip, ipExists := data["ip"]
+		if !ipExists {
+			return fmt.Errorf("each backend must specify an 'ip' field")
+		}
+		if err := validateIPAddress(ip); err != nil {
+			return fmt.Errorf("invalid IP format in backend configuration: %s", err)
+		}
+
+		sourcePort, err := getIntField(data, "source-port")
+		if err != nil {
+			return fmt.Errorf("source-port error: %s", err)
+		}
+
+		targetPort, err := getIntField(data, "target-port")
+		if err != nil {
+			return fmt.Errorf("target-port error: %s", err)
+		}
+
+		healthCheckPort, err := getIntField(data, "health-check-port")
+		if err != nil {
+			return fmt.Errorf("health-check-port error: %s", err)
+		}
+
+		protocol := data["protocol"]
```

Comment:
we should validate the protocol

Reply:
also, if protocol here is not specified it will panic, we should use the `protocol, ok := data["protocol"]` way